<launch>
  <!-- Launch file for vanjee722f VLP-16 LiDAR -->

    <arg name="rviz" default="true" />

    <rosparam command="load" file="$(find fast_lio_localization)/config/vanjee722f.yaml" />

    <param name="map_voxel_size" type="double" value="0.4"/>
    <param name="scan_voxel_size" type="double" value="0.1" />
    <param name="freq_localization" type="double" value="0.5" />
    <param name="localization_th" type="double" value="0.3" />
    <param name="fov" type="double" value="6.28" />
    <param name="fov_far" type="double" value="50.0" />
    <param name="map_file_path" type="string" value="map_20250927_154440_168.pcd" />
    <node pkg="fast_lio_localization" type="fastlio_mapping" name="fast_lio_localization_node" output="screen" /> 

    <arg name="map" default="" />
    <!-- loalization-->
    <!-- <node pkg="fast_lio_localization" type="fastlio_mapping" name="global_localization" output="screen" /> -->

    <!-- 启动坐标变换融合节点 -->
    <node pkg="fast_lio_localization" type="transform_fusion" name="transform_fusion" output="screen">
        <param name="freq_pub_localization" value="50.0"/>
    </node>

    <!-- glbal map-->
    <!-- <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publishe" output="screen"
          args="$(arg map) 5 _frame_id:=/map cloud_pcd:=/map" /> -->
    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization.rviz"/>
    </group>

</launch>

